xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/main/java/org/xwiki/rendering/internal/macro/include/IncludeMacro.java

```diff
@@ -202,7 +202,45 @@ public void setDocumentDisplayer(DocumentDisplayer documentDisplayer)
                 references.pop();
             }
         }
+        
+        // Exclude First Heading 
+        if (parameters.excludeFirstHeading()) {
+            List<HeaderBlock> headingList = result.getBlocks(new ClassBlockMatcher(HeaderBlock.class),
+                    Block.Axes.DESCENDANT);
```

Comment:
I wouldn't use this at all since this is costly for nothing as we're only checking the first block...

Also your logic below is flawed in several places:
* Heading blocks are always since Section blocks, so you just need to check that the first block is a section block and that the fist block in this section block is a heading block. That's just 1 line of code ;)
  * If if it's the case, remove the heading block and the section block.
* Don't throw an exception if it's not the case, just don't do anything.